mangaki/mangaki/models.py

```diff
@@ -240,20 +240,43 @@ class WorkTitle(models.Model):
                             blank=True,
                             db_index=True)
 
+    class Meta:
+        unique_together = ('title', 'language')
+
     def __str__(self):
         return "%s" % self.title
 
+UNK_LANG_VALUE = 'x-unk'
+
+
+class ExtLanguage(models.Model):
+    source = models.CharField(max_length=30)
+    ext_lang = models.CharField(
+        default=UNK_LANG_VALUE,
+        max_length=8,
+        db_index=True
+    )
+    lang = models.ForeignKey('Language')
+
+    class Meta:
+        unique_together = ('ext_lang', 'source')
+
+    def __str__(self):
+        return ('<ExtLanguage: source {}, ext_lang: {}, lang: {}>'
+                .format(self.source,
+                        self.ext_lang,
+                        self.lang.code))
+
 
 class Language(models.Model):
-    anidb_language = models.CharField(max_length=5,
-                                      blank=True,
-                                      db_index=True)
-    iso639 = models.CharField(max_length=2,
-                              unique=True,
-                              db_index=True)
+    code = models.CharField(
+        default=UNK_LANG_VALUE,
+        max_length=10,
+        db_index=True)  # ISO639-1 or custom (for example: x-jat, x-ko, x-ins)
 
     def __str__(self):
-        return "%s : %s" % (self.anidb_language, self.iso639)
+        return ("<Language: {}>"
+                .format(self.code))
```

Comment:
Maybe this is just me (we still don't have a good style guide), but I find this extremely ugly. Why not simply

```python
return "<Language: {}>".format(self.code)
```